src/main/java/trn/logistics/knapsack/service/MaterialService.java

```diff
@@ -17,4 +22,22 @@ public Material getMaterial(String name) {
 
         return materialRepository.findByName(name);
     }
+
+    public Material getMaterial(Long id) {
+        if(id != null) {
+            return materialRepository.findById(id).orElse(null);
+        }
+        return null;
+    }
+
+    public void putMaterial(Material material) {
+        materialRepository.save(material);
+    }
+
+    public ArrayList<Material> getMaterial() {
+        Stream<Material> result = StreamSupport.stream((materialRepository.findAll()).spliterator(), false);
```

Comment:
We could switch the Repositories to extend `JpaRepository` instead of `CrudRepository` to avoid this ugly method completly.
Which would also extend a `CrudRepository`: https://stackoverflow.com/a/14025100